packages/eslint-plugin-example-typed-linting/eslint.config.mjs

```diff
@@ -0,0 +1,21 @@
+import eslint from '@eslint/js';
+import eslintPlugin from 'eslint-plugin-eslint-plugin'
+import tseslint from 'typescript-eslint';
+
+export default tseslint.config(
+    { ignores: ["lib"] },
+    eslint.configs.recommended,
+    ...tseslint.configs.recommendedTypeChecked,
+    eslintPlugin.configs['flat/recommended'],
```

Comment:
Note that this causes a lint error for now: https://github.com/eslint-community/eslint-plugin-eslint-plugin/issues/487 -> https://github.com/eslint-community/eslint-plugin-eslint-plugin/pull/488